modules/sagemaker/sagemaker-templates-service-catalog/templates/model_deploy/github_repo_construct.py

```diff
@@ -0,0 +1,284 @@
+from typing import Any
+
+from aws_cdk import Aws, CustomResource, Duration
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_lambda as lambdafunction
+from constructs import Construct
+
+
+class GitHubRepositoryCreator(Construct):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        github_token_secret_name: str,
+        repo_name: str,
+        repo_description: str,
+        github_owner: str,
+        s3_bucket_name: str,
+        s3_bucket_object_key: str,
+        code_connection_arn: str,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(scope, id, **kwargs)
+
+        github_lambda_func_code = """
```

Comment:
Maybe this could go into a separate file and be loaded into the lambda?

Reply:
As part of next PRs, I will be moving this lambda code into separate common structure so all SageMaker templates can use it. This will be taken care in next PRs.